includes/Context.php

```diff
@@ -192,55 +239,38 @@ public function get_reference_site_url() {
 	 * @return string|false The reference permalink URL or false if post does not exist.
 	 */
 	public function get_reference_permalink( $post = 0 ) {
-		$reference_site_url = $this->get_reference_site_url();
-		$orig_site_url      = home_url();
-
-		// Gets post object. On front area we need to use get_queried_object to get the current post object.
-		if ( ! $post ) {
-			if ( is_admin() ) {
-				$post = get_post();
-			} else {
-				$post = get_queried_object();
-			}
-
-			if ( ! $post instanceof \WP_Post ) {
-				return false;
+		// If post is provided, get URL for that.
+		if ( $post ) {
+			$permalink = get_permalink( $post );
+			if ( false === $permalink ) {
+				return $permalink;
```

Comment:
Personal preference, but I would return `false` here instead. The result is the same but it's more clear IMO.